src/Opdex.Platform.Application/Assemblers/TokenDtoAssembler.cs

```diff
@@ -27,14 +29,23 @@ public async Task<TokenDto> Assemble(Token token)
         var tokenDto = _mapper.Map<TokenDto>(token);
 
         var attributes = await _mediator.Send(new RetrieveTokenAttributesByTokenIdQuery(token.Id), CancellationToken.None);
-        tokenDto.Attributes = attributes.Select(attribute => attribute.AttributeType);
+        var attributeTypes = attributes.Select(attribute => attribute.AttributeType).ToList();
+        tokenDto.Attributes = attributeTypes;
 
         var tokenWrapped = await _mediator.Send(new RetrieveTokenWrappedByTokenIdQuery(token.Id, false), CancellationToken.None);
         if (tokenWrapped is not null)
         {
             tokenDto.WrappedToken = _mapper.Map<WrappedTokenDetailsDto>(tokenWrapped);
         }
 
+        var tokenIsMinedToken = attributeTypes.Contains(TokenAttributeType.Staking);
+        if (tokenIsMinedToken)
+        {
+            var tokenDistribution = await _mediator.Send(new RetrieveLatestTokenDistributionByTokenIdQuery(token.Id), CancellationToken.None);
```

Comment:
Thinking it would be nice to return all previous distributions, they're yearly so even 5 years is only 5 records. What do you think?

Reply:
Updated, so now we're returning all the distributions we have